test/es6/generators-functionality.js

```diff
@@ -1355,16 +1371,16 @@ var tests = [
             assert.areEqual({value: 2, done: true}, g.return(2), "As the return property is missing the yield* just returns as is");
             g = gf();
             assert.areEqual({value: 1, done: false}, g.next(), "Get the first yield value from the inner iterator");
-            assert.throws(function () { g.throw(new ExpectedException()); }, ExpectedException, "As the throw property is missing the yield* just throws as is");
+            assert.throws(function () { g.throw(new ExpectedException()); }, TypeError, "As the throw property is missing a TypeError is thrown");

```

Comment:
Sorry I just noticed that some of the assert.throws you've modified or added do not have the expected exception message string at the end as the (unfortunately optional) fourth argument.

Could you add those in.  Then feel free to squash.

I did a preliminary test run and indeed found that our debugger internal tests need to be updated before this is merged which will take me some time.  I'll get to that next week.